test/al_aws_collector_test.js

```diff
@@ -63,67 +63,102 @@ var formatFun = function (event, context, callback) {
-            return callback(null, params);
-        });
-    
+
+        responseStub = sinon.stub(m_response, 'send').callsFake(
+            function fakeFn(event, context, responseStatus, responseData, physicalResourceId) {
+                context.succeed();
+            });
+
         setAlServiceStub();
     });
 
-    after(function(){
+    afterEach(function(){
         restoreAlServiceStub();
+        responseStub.restore();
     });
 
     it('register success', function(done) {
+        var context = {
+            invokedFunctionArn : colMock.FUNCTION_ARN,
+            succeed : () => {
+                sinon.assert.calledWith(alserviceStub.post, colMock.REG_URL, colMock.REG_AZCOLLECT_QUERY);
+                done();
+            }
+        };
         AlAwsCollector.load().then(function(creds) {
             var collector = new AlAwsCollector(
-            context, 'cwe', AlAwsCollector.IngestTypes.SECMSGS, '1.0.0', creds);
-                collector.register({}, function(error) {
-                done();
-            });
+            context, 'cwe', AlAwsCollector.IngestTypes.SECMSGS, '1.0.0', creds, function() {});
+            collector.register(colMock.REGISTRATION_TEST_EVENT, context, colMock.REG_PARAMS);
         });
     });
     
     it('checkin success', function(done) {
         AlAwsCollector.load().then(function(creds) {
             var collector = new AlAwsCollector(
             context, 'cwe', AlAwsCollector.IngestTypes.SECMSGS,'1.0.0', creds);
-            const mockHealth = {
-                'status':'ok',
-                'details':[],
-                'statistics':[
-                    {'Label':'Invocations','Datapoints':[{'Timestamp':'2017-11-21T16:40:00Z','Sum':1,'Unit':'Count'}]}
-                ]
-            };
-            collector.checkin(mockHealth, function(error) {
+            collector.checkin(colMock.CHECKIN_PARAMS, function(error) {
+                assert.equal(error, undefined);
+                sinon.assert.calledWith(alserviceStub.post, colMock.CHECKIN_URL, colMock.CHECKIN_AZCOLLECT_QUERY);
                 done();
             });
         });
     });
     
     it('deregister success', function(done) {
+        var context = {
+            invokedFunctionArn : colMock.FUNCTION_ARN,
+            succeed : () => {
+                sinon.assert.calledWith(alserviceStub.del, colMock.DEREG_URL);
+                done();
+            }
+        };
         AlAwsCollector.load().then(function(creds) {
             var collector = new AlAwsCollector(
             context, 'cwe', AlAwsCollector.IngestTypes.SECMSGS, '1.0.0', creds);
-            collector.deregister({}, function(error) {
+            collector.deregister(colMock.DEREGISTRATION_TEST_EVENT, context, colMock.DEREG_PARAMS);
+        });
+    });
+
+    it('self update success', function(done) {
+        AWS.mock('Lambda', 'updateFunctionCode', function (params, callback) {
```

Comment:
@kkuzmin I haven't found a way to have mocks in `beforeEach`, and check asserts in the function.

Here https://github.com/dwyl/aws-sdk-mock#use-in-your-tests it is said that it's possible to use `sinon.spy`, but it didn't work for me (the whole function didn't reach to callback I send to `selfUpdate` fun)

If you have an idea how it could done, please let me know, I think having mocks in `beforeEach` would be much cleaner way 👍

Reply:
I think it is ok to have something mocked only for a specific tests. There is no benefit for mocking that for every single test.
You case refactor mocks into a separate function, like in this case, `mockFunctionUpdate()` and call it when necessary.